packages/tpetra/CMakeLists.txt

```diff
@@ -112,21 +112,35 @@ INCLUDE(CMakeCuda.txt)
 # requirements are fulfilled.
 #
 
-SET(${PACKAGE_NAME}_ETI_SCALARS "")
+SET(${PACKAGE_NAME}_ETI_SCALARS "" CACHE STRING "")
```

Comment:
To be safe, you might want to make all of these `FORCE` so they have similar semantics as non-cache vars.  Before, if someone set the cache var `Tpetra_ETI_SCALARS`, then the non-cache set statement:

```cmake
SET(${PACKAGE_NAME}_ETI_SCALARS_NO_ORDS "")
```

would shadow that cache var and basically ignore it.

But if you set this to:

```
SET(${PACKAGE_NAME}_ETI_SCALARS "" CACHE STRING "" FORCE)
```

then it will write over an existing cache var.

Note that changing the local var in this CMakeLists.txt file as a cache var has the potential to break backwards compatibility but it is very unlikely to do so.